### PR TITLE
Feature/get agent

### PR DESCRIPTION
**Pull request by AodhAnd**
* Added tests for get agent client's function
* Implemented get agent client function

### CODE REVIEW

File: craftai/client.py [hunk 1]
```diff
@@ -63,6 +63,8 @@ def create_agent(self, model, agent_id=""):
             raise CraftAINotFoundError(r.text)
         if r.status_code == requests.codes.bad_request:
             raise CraftAIBadRequestError(r.text)
+        if r.status_code == requests.codes.unauthorized:
```
Comment (matthieu-boussard) [inline craftai/client.py]: use a switch

File: craftai/client.py [hunk 2]
```diff
@@ -72,7 +74,35 @@ def create_agent(self, model, agent_id=""):
         return agent
 
     def get_agent(self, agent_id):
-        pass
+        if (not agent_id) or (not isinstance(agent_id, six.string_types)):
```
Comment (matthieu-boussard) [inline craftai/client.py]: use agent_id is None instead of not agent_id

File: craftai/client.py [hunk 3]
```diff
@@ -72,7 +74,35 @@ def create_agent(self, model, agent_id=""):
...
+            self.cfg.get("url"),
+            "api/",
+            self.cfg.get("owner"),
+            "/agents/",
+            agent_id)
+        )
+
+        r = requests.get(url, headers=headers)
+
+        if r.status_code == requests.codes.not_found:
```
Comment (matthieu-boussard) [inline craftai/client.py]: switch (or make a function use by all routes)

File: craftai/client.py [hunk 4]
```diff
@@ -94,6 +124,8 @@ def delete_agent(self, agent_id):
 
         if r.status_code == requests.codes.bad_request:
             raise CraftAIBadRequestError(r.text)
+        if r.status_code == requests.codes.unauthorized:
+            raise CraftAICredentialsError(r.text)
```
Comment (matthieu-boussard) [inline craftai/client.py]: yes definitely make a function

File: craftai/client.py [hunk 5]
```diff
@@ -72,7 +74,35 @@ def create_agent(self, model, agent_id=""):
         return agent
 
     def get_agent(self, agent_id):
-        pass
+        if (not agent_id) or (not isinstance(agent_id, six.string_types)):
+            raise CraftAIBadRequestError("agent_id has to be a string")
+
+        headers = {}
```
Comment (matthieu-boussard) [inline craftai/client.py]: (as we said) move the header in the client class

File: craftai/client.py [hunk 6]
```diff
@@ -72,7 +74,35 @@ def create_agent(self, model, agent_id=""):
...
+        headers["Authorization"] = "Bearer " + self.cfg.get("token")
+        headers["Accept"] = "application/json"
+        url = "".join((
```
Comment (matthieu-boussard) [inline craftai/client.py]: use placeholder insead of join

File: craftai/client.py [hunk 7]
```diff
@@ -72,7 +74,35 @@ def create_agent(self, model, agent_id=""):
...
+            self.cfg.get("url"),
+            "api/",
+            self.cfg.get("owner"),
```
Comment (matthieu-boussard) [inline craftai/client.py]: you can use cfg["owner"]

File: craftai/client.py (hunk 2 above)
Comment (AodhAnd) [inline craftai/client.py]: write explicitly the test for empty strings (not agent_id is not explicit)

File: craftai/client.py (hunk 1 above)
Comment (AodhAnd) [inline craftai/client.py]: No switch in python, but error checking will be factorized in a separated function.

**Pull request merged**